test/modules/geo-layers/tile-layer/tile-2d-header.spec.js

```diff
@@ -65,3 +67,78 @@ test('Tile2DHeader#Abort quickly', async t => {
   t.notOk(requestScheduler.requestMap.has(tile), 'Scheduler deletes tile on abort');
   t.end();
 });
+
+const LOCAL_TILE_COORDS = [
+  {
+    type: 'Feature',
+    geometry: {
+      type: 'Polygon',
+      coordinates: [
+        [
+          [0.80908203125, 0.8935546875],
+          [0.8095703125, 0.89404296875],
+          [0.80908203125, 0.89404296875],
+          [0.80908203125, 0.8935546875]
+        ]
+      ]
+    }
+  }
+];
+
+test('Tile2DHeader#dataInWorldCoordinates', async t => {
+  const [loader, tile] = createTile2DHeaderWithMockedData(LOCAL_TILE_COORDS);
+
+  t.notOk(tile.isLoaded, 'Data has not been loaded yet');
+  t.deepEqual(await tile.dataInWorldCoordinates, await loader);
+  t.ok(tile.isLoaded, 'Data has been loaded');
+  t.deepEqual(
+    await tile.dataInWorldCoordinates,
+    tile.content,
+    'Correctly returns no-transformed data'
+  );
+
+  t.end();
+});
+
+const TRANSFORM_TO_WORLD_VIEWPORT = new WebMercatorViewport({
+  latitude: 0,
+  longitude: 0,
+  zoom: 1
+});
+
+const TRANSFORM_TO_WORLD_BBOX = {west: -180, north: 85.0511287798066, east: 0, south: 0};
+
+test('Tile2DHeader#transformToWorld', async t => {
+  const [loader, tile] = createTile2DHeaderWithMockedData(LOCAL_TILE_COORDS);
+
+  tile.bbox = TRANSFORM_TO_WORLD_BBOX;
+  tile.transformToWorld = ([object]) =>
+    transformTileCoordsToWGS84(object, tile, TRANSFORM_TO_WORLD_VIEWPORT);
+
+  t.notOk(tile.isLoaded, 'Data has not been loaded yet');
+  t.deepEqual(await tile.dataInWorldCoordinates, await loader);
+  t.is(tile._transformToWorldDataCache, null, 'Data cache is empty');
+  t.ok(tile.isLoaded, 'Data has been loaded');
+  t.deepEqual(
+    await tile.dataInWorldCoordinates,
+    tile._transformToWorldDataCache,
+    'Correctly returns transformed data'
+  );
+
+  t.end();
+});
+
+function createTile2DHeaderWithMockedData(data) {
+  const requestScheduler = new RequestScheduler({throttleRequests: true, maxRequests: 1});
+
+  const getTileData = () => Promise.resolve(data);
+  const onTileLoad = () => null;
+  const onTileError = () => null;
+
+  const tile = new Tile2DHeader({onTileLoad, onTileError});
+  tile.isSelected = true;
+
+  const loader = tile._loadData(getTileData, requestScheduler);
```

Comment:
I would keep calling `tile._loadData`, even if it's a private method, since the [other implemented tests](https://github.com/visgl/deck.gl/blob/013380268dcf3fb4a2b0d4000f2ac2e464c8935f/test/modules/geo-layers/tile-layer/tile-2d-header.spec.js#L41) are using this method, so would need to modify them..